service/rpc/das.go

```diff
@@ -0,0 +1,52 @@
+package rpc
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/celestiaorg/celestia-node/das"
+)
+
+const (
+	dasStateEndpoint = "/daser"
+)
+
+// DasStateResponse encompasses the fields returned in response
+// to a `/daser` request.
+type DasStateResponse struct {
+	IsSampling   bool `json:"is_sampling"`
+	IsCatchingUp bool `json:"is_catching_up"`
+
+	LatestSampledHeight uint64 `json:"latest_sampled_height"`
+	LatestSampledWidth  uint64 `json:"latest_sampled_square_width"`
+
+	Err map[string]das.SampleError `json:"errors"`
+}
+
+func (h *Handler) handleDASStateRequest(w http.ResponseWriter, r *http.Request) {
+	dasState := new(DasStateResponse)
+	sampleState := h.das.SampleRoutineState()
+	catchUpState := h.das.CatchUpRoutineState()
```

Comment:
What if das state will change between fetching the state and returning the response?